### PR TITLE
deprecate C*KVSConfig.safetyDisabled

### LINKED ISSUE #2004: Opt Out of Specific CassandraVerifier Checks
See PDS-53324 for an example of this.

CassandraVerifier does a various number of sanity checks around your atlas config, the actual configuration of the cassandra keyspace, and various other aspects of the cassandra cluster. In some cases deployments fail one of these checks because they're in an unideal layout (but correct for their limiting circumstances). The most common of these is having greater than 3 nodes with RF3 all in the same rack due to hardware constraints on-premise.

It would be great if we could override a single check and still opt-in to the rest of them in these cases. Currently setting `safetyDisabled` to `true` disables ALL the checks.

Also CassandraVerifier could generally use a refactor/cleanup.

### PR DESCRIPTION
**Pull request by clockfort**
fixes #2004

**Concerns (what feedback would you like?)**:
Not sure if we really need to go through a round of deprecation; these are very rarely used and we don't even have a really great way outside of release notes to actually tell users they are using deprecated config options. I doubt people will notice until we actually hard-drop any options they're using that are currently deprecated.

**Priority (whenever / two weeks / yesterday)**:
whenever

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2024)

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -42,9 +42,10 @@ develop
     *    - Type
          - Change
 
-    *    -
-         -
-
+    *    - |userbreak|
```
Comment (jeremyk-91) [inline docs/source/release_notes/release-notes.rst]: This seems a little harsh - we should probably use `|deprecated|`.

We should document the params a little more clearly, but let's get this PR in first which will be useful for several customers

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java [hunk 2]
```diff
@@ -271,38 +271,39 @@ private static void checkAndSetReplicationFactor(
...
 
-    private static void sanityCheckReplicationFactor(KsDef ks, int desiredRf, boolean safetyDisabled, Set<String> dcs) {
+    private static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
         Map<String, String> strategyOptions = Maps.newHashMap(ks.getStrategy_options());
         for (String dc : dcs) {
             if (strategyOptions.get(dc) == null) {
                 logErrorOrThrow("The datacenter for this cassandra cluster is invalid. "
                         + " failed dc: " + dc
-                        + "  strategyOptions: " + strategyOptions, safetyDisabled);
+                        + "  strategyOptions: " + strategyOptions, config.ignoreBadDatacenterConfigurationChecks());
```
Comment (jeremyk-91) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java]: Should this be `config.safetyDisabled() || config.ignoreBadDatacenterConfigurationChecks()`?

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java [hunk 3]
```diff
@@ -268,50 +275,48 @@ private static void checkAndSetReplicationFactor(
...
+        Set<String> dcs = sanityCheckDatacenters(client, config);
+        sanityCheckReplicationFactor(ks, config, dcs);
     }
 
-    private static void sanityCheckReplicationFactor(KsDef ks, int desiredRf, boolean safetyDisabled, Set<String> dcs) {
+    private static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
         Map<String, String> strategyOptions = Maps.newHashMap(ks.getStrategy_options());
         for (String dc : dcs) {
             if (strategyOptions.get(dc) == null) {
                 logErrorOrThrow("The datacenter for this cassandra cluster is invalid. "
```
Comment (jeremyk-91) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java]: Might be worth mentioning the flag to set to override said checks? Though I'm not sure what kind of use cases might be legitimate for this :/

Comment (clockfort) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java]: yeah, this is why I didn't mention it for this specific one >_>

**Pull request merged**